moveit_ros/planning/planning_pipeline/include/moveit/planning_pipeline/planning_pipeline_middleware_handle.hpp

```diff
@@ -0,0 +1,132 @@
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Anthony Baker */
+
+#pragma once
+#include <moveit/planning_pipeline/planning_pipeline.h>
+#include <rclcpp/rclcpp.hpp>
+#include <string>
+
+namespace planning_pipeline
+{
+class PlanningPipelineMiddlewareHandle : public PlanningPipeline::MiddlewareHandle
+{
+public:
+  PlanningPipelineMiddlewareHandle(const rclcpp::Node::SharedPtr& node);
+
+  /**
+   * @brief     Create a display path publisher
+   * @param[in] topic the topic
+   */
+  void createDisplayPathPublisher(const std::string& topic, bool flag) override;
+
+  /**
+   * @brief     Create a received request publisher
+   * @param[in] topic the topic
+   */
+  void createReceivedRequestPublisher(const std::string& topic, bool flag) override;
+
+  /**
+   * @brief     Create a contacts publisher
+   * @param[in] topic the topic
+   */
+  void createContactsPublisher(const std::string& topic, bool flag) override;
+
+  std::string getContactsTopicName() const override;
+
+  /**
+   * @brief     Reset a display path publisher
+   */
+  void resetDisplayPathPublisher() override;
+
+  /**
+   * @brief     Reset a received request publisher
+   */
+  void resetReceivedRequestPublisher() override;
+
+  /**
+   * @brief     Reset a contacts publisher
+   */
+  void resetContactsPublisher() override;
+
+  bool has_parameter(const std::string& name) const override;
+
+  void get_parameter(const std::string& pluginName, std::string& parameter) const override;
+
+  /**
+   * @brief     Create a planner plugin
+   * @param[in] name plugin name
+   */
+  void createPlannerPlugin(const std::string& name, const moveit::core::RobotModelConstPtr& robot_model) override;
+
+  const planning_interface::PlannerManagerPtr& getPlannerManager() const override;
+
+  /**
+   * @brief     Create a adapter plugin
+   * @param[in] name plugin name
+   */
+  void createAdapterPlugins(const std::vector<std::string>& name) override;
+
+  bool plan(const planning_scene::PlanningSceneConstPtr& planning_scene,
+            const planning_interface::MotionPlanRequest& req, planning_interface::MotionPlanResponse& res,
+            std::vector<std::size_t>& adapter_added_state_index) override;
+
+  /**
+   * @brief     Create a contacts publisher
+   * @param[in] topic the topic
+   */
+  void terminatePlannerPlugin() override;
+
+  void publishReceivedRequest(const planning_interface::MotionPlanRequest& req) override;
+  void publishContactsMarkerArray(const visualization_msgs::msg::MarkerArray& markerArray) override;
+  void publishDisplayPathTrejectory(const moveit_msgs::msg::DisplayTrajectory& displayTrajectory) override;
```

Comment:
Blanket comment: please add documentation for every function that is missing it and check that every parameter is included.